client/lib/signup/step-actions/index.js

```diff
@@ -411,15 +428,21 @@ function processItemCart(
 
 	const userLoggedIn = isUserLoggedIn( reduxStore.getState() );
 
+	const flowToCheck = stepData.flowName || stepData.lastKnownFlow;
+	const skipSitesAndUsersFetch =
+		isEnabled( 'signup/setup-site-after-checkout' ) &&
```

Comment:
this could use a comment explaining that we will perform the check on the theme select page in this case

Reply:
Agreed!